### PR TITLE
Launching fighters uncloaks stealth carrier

### PR DESCRIPTION
**Pull request by agrrr3**
This PR intends to fix the "feature" that stealth carriers (ships with launch bays, hangars and no ShortRange weapons) are basically invincible as long as the ship stealth is higher than enemy's detection.

This PR makes launching fighters on par with firing a weapon from a stealthed ship - it makes the ship known to the enemies.

This is WIP. The carriers correctly unstealth but there is no event shown in the combat report. I do not grok the event code fully yet so I need some assistance/review how to proceed with the implementation.

### CODE REVIEW

**Comment by agrrr3** [comment]
Ok, implementation now works as intended - SitRep that foreign empires can detect . Besides stealth carrriers I also checked for attack carriers (having ShortRange weapon besides fighters).

I had a look at the failing CI reports but those looked unrelated - so I would say the fails are bleeps.

So... ready-for-review.

File: combat/CombatSystem.cpp [hunk 1]
```diff
@@ -1746,6 +1747,31 @@ namespace {
                                launches_event);
 
                 DebugLogger(combat) << "Attacker: " << attacker->Name();
+
+                // Set launching carrier as at least basically visible to other empires.
+                if (launches_event) {
```
Comment (geoffthemedio) [inline combat/CombatSystem.cpp]: does this test do anything?

Comment (agrrr3) [reply combat/CombatSystem.cpp]: It should have tested that there were actually launch events

**Comment by agrrr3** [comment]
#2569 should fix the CI errors.

**Pull request merged**